python/tests/remote/test_remote_platform.py

```diff
@@ -0,0 +1,125 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+import pytest
+import os, math
+import cudaq
+
+def has_rest_server():
+    try:
+        import subprocess
+        subprocess.check_output(["which", "cudaq_rest_server"])
+        return True
+    except:
+        return False
+
+skipIfNoRestServer = pytest.mark.skipif(
+    not (has_rest_server()), reason="cudaq_rest_server not available")
+
+num_qpus = 3
+
+@pytest.fixture(scope="session", autouse=True)
+@skipIfNoRestServer
+def startUpMockServer():
+    cudaq.set_target("remote-sim", auto_launch=str(num_qpus))
+    yield
+    cudaq.reset_target()
+
+@skipIfNoRestServer
+def test_setup():
+    target = cudaq.get_target()
+    numQpus = target.num_qpus()
+    assert numQpus == num_qpus
+
+@skipIfNoRestServer
+def test_sample():
+    kernel = cudaq.make_kernel()
+    qubits = kernel.qalloc(2)
+    kernel.h(qubits[0])
+    kernel.cx(qubits[0], qubits[1])
+    kernel.mz(qubits)
+    counts = cudaq.sample(kernel)
+    print(counts)
+
+    assert len(counts) == 2
+    assert "00" in counts
+    assert "11" in counts
+
+    future = cudaq.sample_async(kernel)
+    counts = future.get()
+    print(counts)
+    assert len(counts) == 2
+    assert "00" in counts
+    assert "11" in counts
+
+@skipIfNoRestServer
+def test_observe():
+    # Create the parameterized ansatz
+    kernel, theta = cudaq.make_kernel(float)
+    qreg = kernel.qalloc(2)
+    kernel.x(qreg[0])
+    kernel.ry(theta, qreg[1])
+    kernel.cx(qreg[1], qreg[0])
+    from cudaq import spin
+    # Define its spin Hamiltonian.
+    hamiltonian = (5.907 - 2.1433 * spin.x(0) * spin.x(1) -
+                   2.1433 * spin.y(0) * spin.y(1) + 0.21829 * spin.z(0) -
+                   6.125 * spin.z(1))
+
+    res = cudaq.observe(kernel, hamiltonian, 0.59)
+    print("Energy =", res.expectation())
+    assert abs(res.expectation() + 1.748794) < 0.01
```

Comment:
Suggestion - replace magic number `1.748794` with named variable. Same for tolerance `0.01`.

Reply:
Good point. Fixed as suggested.